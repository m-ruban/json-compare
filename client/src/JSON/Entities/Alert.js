import React from 'react'
import {Badge} from 'reactstrap'

const Alert = ({id, alert, showAlerts, showRequired}) => {

  let node = null
  if ((showAlerts || showRequired) && alert) {

    let show = false
    if (showAlerts) show = alert.res === 'diff'
    if (showRequired && !show) show = alert.res === 'req'

    if (show) {
      node = <Badge color={alert.res} pill>!</Badge>
    }

  }

  return node
}

export default Alert
