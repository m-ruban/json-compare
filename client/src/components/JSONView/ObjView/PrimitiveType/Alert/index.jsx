import React from 'react';
import PropTypes from 'prop-types';
import { COMPARE_REQ, COMPARE_DIFF } from 'constants/compare';
import Mark from 'components/Mark';

const Alert = ({ alert, showAlerts, showRequired }) => {
    let node = null;
    if ((showAlerts || showRequired) && alert) {
        let show = false;
        if (showAlerts) show = alert.res === COMPARE_DIFF;
        if (showRequired && !show) show = alert.res === COMPARE_REQ;

        if (show) {
            node = <Mark type={alert.res} text="!" pill />;
        }
    }

    return node;
};

Alert.propTypes = {
    alert: PropTypes.object,
    showAlerts: PropTypes.bool,
    showRequired: PropTypes.bool,
};

export default Alert;
