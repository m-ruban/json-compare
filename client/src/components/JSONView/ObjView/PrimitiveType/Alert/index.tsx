import React from 'react';

import Mark from 'components/Mark';
import { COMPARE_DIFF, COMPARE_REQ } from 'constants/compare';

interface IAlert {
    res: string;
    scalar: boolean;
}

interface IAlertProps {
    alert: IAlert;
    showAlerts: boolean;
    showRequired: boolean;
}

const Alert = ({ alert, showAlerts, showRequired }: IAlertProps) => {
    let node = null;
    if ((showAlerts || showRequired) && alert) {
        let show = false;
        if (showAlerts) { show = alert.res === COMPARE_DIFF; }
        if (showRequired && !show) { show = alert.res === COMPARE_REQ; }

        if (show) {
            node = <Mark type={alert.res} text="!" pill />;
        }
    }

    return node;
};

export default Alert;
