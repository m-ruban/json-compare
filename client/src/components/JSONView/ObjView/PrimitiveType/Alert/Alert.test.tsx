import React from 'react';

import Alert from 'components/JSONView/ObjView/PrimitiveType/Alert';
import { render } from 'enzyme';
import alerts from 'tests/mocks/alerts';

describe('<Alert />', () => {
    it('Renders correctly', () => {
        const props = { alert: alerts.alerts.get('test'), showAlerts: false, showRequired: false };
        const output = render(<Alert {...props} />);
        expect(output).toMatchSnapshot();
    });
});
