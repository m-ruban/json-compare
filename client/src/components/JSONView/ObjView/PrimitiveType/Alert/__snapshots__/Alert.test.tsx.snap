// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Renders correctly 1`] = `null`;
